src/PaymentGateways/PayPalCommerce/DataTransferObjects/PayPalWebhookHeaders.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace Give\PaymentGateways\PayPalCommerce\DataTransferObjects;
+
+use HttpHeaderException;
+
+class PayPalWebhookHeaders {
+	/**
+	 * @since 2.9.0
+	 * @var string
+	 */
+	public $transmissionId;
+
+	/**
+	 * @since 2.9.0
+	 * @var string
+	 */
+	public $transmissionTime;
+
+	/**
+	 * @since 2.9.0
+	 * @var string
+	 */
+	public $transmissionSig;
+
+	/**
+	 * @since 2.9.0
+	 * @var string
+	 */
+	public $certUrl;
+
+	/**
+	 * @since 2.9.0
+	 * @var string
+	 */
+	public $authAlgo;
+
+	/**
+	 * This grabs the headers from the webhook request to be used in the signature verification
+	 *
+	 * A strange thing here is that the headers are inconsistent between live and sandbox mode, so this also checks for
+	 * both forms of the headers (studly case and all caps).
+	 *
+	 * @since 2.9.0
+	 *
+	 * @param array $headers
+	 *
+	 * @return self
+	 * @throws HttpHeaderException
+	 */
+	public static function fromHeaders( array $headers ) {
+		$headerKeys = [
+			'transmissionId'   => 'Paypal-Transmission-Id',
+			'transmissionTime' => 'Paypal-Transmission-Time',
+			'transmissionSig'  => 'Paypal-Transmission-Sig',
+			'certUrl'          => 'Paypal-Cert-Url',
+			'authAlgo'         => 'Paypal-Auth-Algo',
+		];
+
+		$payPalHeaders = new self();
+		$missingKeys   = [];
+		foreach ( $headerKeys as $property => $key ) {
+			if ( ! isset( $headers[ $key ] ) ) {
+				$key = strtoupper( $key );
+			}
+
+			if ( isset( $headers[ $key ] ) ) {
```

Comment:
This seems like it should be the else statement for the condition on line 63, since it is the opposite of that condition.

Reply:
Hah! Yeah, I can see why it looks like that. The reason it's not the case is because `$key` is changing within the first condition. That is, if the `$key` is not found then change it to all uppercase and try again.